lib/cc/cli/engines.rb

```diff
@@ -9,6 +9,12 @@ module Engines
       autoload :Install, "cc/cli/engines/install"
       autoload :List, "cc/cli/engines/list"
       autoload :Remove, "cc/cli/engines/remove"
+
+      REGISTRY_LIST ||= CC::Analyzer::EngineRegistry.new.list
```

Comment:
I like the suggestion of using `CC::Analyzer::EngineRegistry.exists?(engine_name)`.